modules/3d-tiles/src/tz3/tz3-archive.ts

```diff
@@ -0,0 +1,82 @@
+import md5 from 'md5';
+import {FileProvider, parseZipLocalFileHeader, HashElement, findBin} from '@loaders.gl/zip';
+import {DeflateCompression} from '@loaders.gl/compression';
+
+type CompressionHandler = (compressedFile: ArrayBuffer) => Promise<ArrayBuffer>;
+
+/**
+ * Handling different compression types in zip
+ */
+const COMPRESSION_METHODS: {[key: number]: CompressionHandler} = {
+  /** No compression */
+  0: async (compressedFile) => compressedFile,
+
+  /** Deflation */
+  8: async (compressedFile) => {
+    const compression = new DeflateCompression({raw: true});
+    const decompressedData = await compression.decompress(compressedFile);
+    return decompressedData;
+  }
+};
```

Comment:
Does something like this work? Compression module does provide a `NoCompression` class with the same interface.
```suggestion
const COMPRESSION_METHODS: {[key: number]: CompressionHandler} = {
  /** No compression */
  0: new NoCompression().decompress,
  /** Deflation */
  8: new DeflateCompression({raw: true}).decompress
};
```